.github/workflows/publish_storybook.yml

```diff
@@ -0,0 +1,59 @@
+# Workflow name
+name: Build and Publish Storybook to GitHub Pages
+
+on:
+  # Event for the workflow to run on
+  push:
+    branches:
+      - "main" # Replace with the branch you want to deploy from
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# List of jobs
+jobs:
+  retrieve_pages_url:
+    runs-on: ubuntu-latest
+    outputs:
+      pages_url: ${{ steps.get_url.outputs.url }}
+    steps:
+      - id: get_url
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          url=$(gh api "repos/$GITHUB_REPOSITORY/pages" --jq '.html_url')
+          echo "url=$url" >> "$GITHUB_OUTPUT"
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.build-publish.outputs.page_url }}
+    runs-on: ubuntu-latest
+    needs: retrieve_pages_url
+    # Job steps
+    steps:
+      # Manual Checkout
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
why are we fetching history

Reply:
Copypaste from storybook docs example, from quick glance doesn't seem to actually do anything with it so i'll remove it.